### PR TITLE
Re-enable tests to start kernel from kernelspec

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #5015

### CODE REVIEW

File: src/client/datascience/kernel-launcher/kernelDaemon.ts [hunk 1]
```diff
@@ -125,7 +129,10 @@ export class PythonKernelDaemon extends BasePythonDaemon implements IPythonKerne
                     // Don't call this.subject.error, as that can only be called once (hence can only be handled once).
                     // Instead log this error & pass this only when the kernel dies.
                     stdErr += out.out;
-                    traceWarning(`Kernel ${this.proc.pid} as possibly died, StdErr from Kernel Process ${out.out}`);
+                    // If we have requested for kernel to be killed, don't raise kernel died error.
+                    if (!this.killed) {
+                        traceWarning(`Kernel ${this.proc.pid} as possibly died, StdErr from Kernel Process ${out.out}`);
```
Comment (DonJayamanne) [inline src/client/datascience/kernel-launcher/kernelDaemon.ts:134]: We were unnecessarily throwing & logging errors when we were manually killing he kernels.

File: src/client/datascience/kernel-launcher/kernelDaemon.ts [hunk 2]
```diff
@@ -135,6 +142,12 @@ export class PythonKernelDaemon extends BasePythonDaemon implements IPythonKerne
         );
 
         // If the daemon dies, then kernel is also dead.
-        this.closed.catch((error) => this.subject.error(new PythonKernelDiedError({ error, stdErr })));
+        this.closed.catch((error) => {
+            // If we have requested for kernel to be killed, don't raise kernel died error.
+            if (this.killed) {
+                return;
+            }
```
Comment (DonJayamanne) [inline src/client/datascience/kernel-launcher/kernelDaemon.ts:149]: We were unnecessarily throwing & logging errors when we were manually killing he kernels.

File: src/client/datascience/kernel-launcher/kernelDaemon.ts [hunk 3]
```diff
@@ -107,6 +107,10 @@ export class PythonKernelDaemon extends BasePythonDaemon implements IPythonKerne
         );
         let stdErr = '';
         this.connection.onNotification(KernelDiedNotification, (output) => {
+            // If we have requested for kernel to be killed, don't raise kernel died error.
+            if (this.killed) {
```
Comment (DonJayamanne) [inline src/client/datascience/kernel-launcher/kernelDaemon.ts:111]: We were unnecessarily throwing & logging errors when we were manually killing he kernels.

File: src/test/datascience/kernelLauncher.vscode.test.ts [hunk 4]
```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+'use strict';
+
+import { assert, use } from 'chai';
```
Comment (DonJayamanne) [inline src/test/datascience/kernelLauncher.vscode.test.ts:5]: Moved to VS Code tests to use the real ioc, tests are faster here
The tests were pretty slow in functional tests as for every single test we were starting the IOC and a lot of code was running in the background to load styles, etc... (lots in the logs for every test).

This seem to have fixed the test as well - i just moved it to run as a VS Code test & works.

File: src/client/datascience/kernel-launcher/kernelDaemon.ts (hunk 3 above)
Comment (IanMatthewHuff) [reply src/client/datascience/kernel-launcher/kernelDaemon.ts:111]: Is this the error code -9 warning?

**Review by IanMatthewHuff — APPROVED** [review]

**Review by DavidKutu — APPROVED** [review]

**Review by rchiodo — APPROVED** [review]

**Pull request merged**